tests/ml/dataset_tests.py

```diff
@@ -35,6 +41,48 @@ def test_create_dataset(self):
     ds = CsvDataSet(file_pattern='some/file', schema=schema_str)
     self.assertEqual(json_schema, ds.schema)
 
+  def test_sample(self):
+    tmp_dir = tempfile.mkdtemp()
+    try:
+      json_schema = TestCsvDataSet._create_json_schema()
+      TestCsvDataSet._create_csv_files(tmp_dir, 'data', 3)
+      ds = CsvDataSet(file_pattern=os.path.join(tmp_dir, 'data*'), schema=json_schema)
+
+      df = ds.sample(5)
+      self.assertEqual(5, len(df))
```

Comment:
Also assert the following?

- the 5 samples belong to csv file,
- they are different (so we may need to generate a unique key for each row)
- Returned type is dataframe